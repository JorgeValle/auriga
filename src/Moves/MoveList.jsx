import React from 'react';

// Components should represent one piece of data model
class MoveList extends React.Component {

  // Components should only update their own state
  render() {
    return (
      <table class="move-list">
        <thead class="move-list-header">
          <tr>
            <th class="grey">#</th>
            <th class="white">White</th>
            <th class="black">Black</th>
          </tr>
        </thead>
        <tbody>
          <tr className="move-1">
            <th class="grey">1</th>
            <td>a4</td>
            <td class="black">c5</td>
          </tr>
          <tr className="move-2">
            <th class="grey">2</th>
            <td>a1</td>
            <td class="black">c8</td>
          </tr>
          <tr className="move-3">
            <th class="grey">3</th>
            <td>b4</td>
            <td class="black">e5</td>
          </tr>
        </tbody>
      </table>
    );
  }
}

export default MoveList;