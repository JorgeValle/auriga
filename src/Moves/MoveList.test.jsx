import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import MoveList from './MoveList.jsx';

configure({
  adapter: new Adapter()
});

describe('<MoveList />', () => {

  // We shallow render to make sure changes in children don't cause cascading fails
  it('shallow renders', () => {
    const wrapper = shallow(<MoveList />);
  });

  // Snapshot matching
  it('matches snapshot', () => {
    const wrapper = shallow(<MoveList />);
    expect(wrapper).toMatchSnapshot();
  });

});

