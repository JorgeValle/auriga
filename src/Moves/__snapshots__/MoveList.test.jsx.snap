// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoveList /> matches snapshot 1`] = `ShallowWrapper {}`;
